### PR TITLE
add streaming support

### PR DESCRIPTION
**Pull request by robertkowalski**
this adds streaming support for grenache.

the server emits a `stream` event before the request is
parsed and buffered. with the setting `disableBuffered: true`
we can then disable the convenience auto-parsing and allow more
granular control on the serverside with the req and response
objects / streams. an object called `meta` is emitted, with
`isStream (bool)`, indicating a streaming client. in case
of a stream it also contains req-id and key, which are parsed
from the http header, as the stream is reserved for the (binary)
data.

when a stream is sent with the `strean` method, it
will add the grenache-specific req-id and key to the request
headers. if we want to send additional, grenache-specific data,
the arguments and actions, we can add those in our request:

```js
const req = pst.stream('rpc_stream', {
  timeout: 10000,
  headers: {
    _a: 'uploadPublic',
    _ar: { foo: 'bar' }
  }
})
```

### CODE REVIEW

File: test/helper.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    }
+  ], () => {
+    cb(null, [ grape1, grape2 ])
+  })
+}
+
+exports.killGrapes = killGrapes
+function killGrapes (grapes, done) {
+  grapes[0].stop((err) => {
+    if (err) throw err
```
Comment (quadramadery) [inline test/helper.js:36]: shouldn't this be `done(err)` instead of `throw err` to fail the test?

File: test/before-after.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const { bootTwoGrapes, killGrapes } = require('./helper')
+
+let grapes
+function setupHooks (cb) {
+  beforeEach(function (done) {
+    this.timeout(20000)
+    bootTwoGrapes((err, g) => {
+      if (err) throw err
+      grapes = g
+      grapes[0].once('announce', () => {
```
Comment (quadramadery) [inline test/before-after.js:14]: `() => { done() }` could be written simply as `done`

File: package.json [hunk 3]
```diff
@@ -11,6 +11,9 @@
     "micro-services"
   ],
   "dependencies": {
+    "async": "^2.6.0",
```
Comment (quadramadery) [inline package.json:14]: isn't mocha missing in devDependencies and maybe even scripts.test?

File: test/helper.js (hunk 1 above)
Comment (robertkowalski) [reply test/helper.js:36]: yes, thats possible too, in both cases the test will fail.

File: test/before-after.js (hunk 2 above)
Comment (robertkowalski) [reply test/before-after.js:14]: yes, i like the callback added, so its fast to hook in other stuff (temporarily) for testing or debugging and i don't need to add a new callback first

File: package.json (hunk 3 above)
Comment (robertkowalski) [reply package.json:14]: yes, added!

**Comment by prdn** [comment]
Beautiful!

**Pull request merged**